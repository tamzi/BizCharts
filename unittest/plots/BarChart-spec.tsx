import React from 'react';
import BarChart from '../../src/plots/BarChart';
import { render, cleanup } from '@testing-library/react';

const MOCK_DATA = [
  { type: '家具家电', sales: 38 },
  { type: '粮油副食', sales: 52 },
  { type: '生鲜水果', sales: 61 },
  { type: '美容洗护', sales: 145 },
  { type: '母婴用品', sales: 48 },
  { type: '进口食品', sales: 38 },
  { type: '食品饮料', sales: 38 },
  { type: '家庭清洁', sales: 38 },
];

describe('Plots-BarChart', () => {
  test('基础条形图', () => {
    let chart = null;
    render(<BarChart
      data={MOCK_DATA}
      xField="sales"
      yField="type"
      colorField="type"
      onGetG2Instance={
        (c) => chart = c
      }
    />)
    expect(chart.options).toMatchSnapshot();
  })
  test('colorField --> seriesField', () => {
    let chart = null;
    render(<BarChart
      data={MOCK_DATA}
      xField="sales"
      yField="type"
      colorField="type"
      onGetG2Instance={
        (c) => chart = c
      }
    />)
    expect(chart.options.seriesField).toBe('type');
    cleanup();
  });
})
