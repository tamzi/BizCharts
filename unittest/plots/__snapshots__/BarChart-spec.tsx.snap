// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Plots-BarChart 基础条形图 1`] = `
Object {
  "animation": true,
  "appendPadding": Array [
    10,
    5,
    10,
    10,
  ],
  "autoFit": true,
  "barWidthRatio": 0.6,
  "colorField": "type",
  "data": Array [
    Object {
      "sales": 38,
      "type": "家具家电",
    },
    Object {
      "sales": 52,
      "type": "粮油副食",
    },
    Object {
      "sales": 61,
      "type": "生鲜水果",
    },
    Object {
      "sales": 145,
      "type": "美容洗护",
    },
    Object {
      "sales": 48,
      "type": "母婴用品",
    },
    Object {
      "sales": 38,
      "type": "进口食品",
    },
    Object {
      "sales": 38,
      "type": "食品饮料",
    },
    Object {
      "sales": 38,
      "type": "家庭清洁",
    },
  ],
  "interactions": Array [
    Object {
      "type": "active-region",
    },
  ],
  "legend": Object {
    "position": "right-top",
    "reversed": true,
  },
  "marginRatio": 0.03125,
  "renderer": "canvas",
  "seriesField": "type",
  "tooltip": Object {
    "offset": 20,
    "reversed": true,
    "shared": true,
    "showMarkers": false,
  },
  "xAxis": Object {
    "label": Object {
      "autoHide": Object {
        "cfg": Object {
          "minGap": 6,
        },
        "type": "equidistance",
      },
      "autoRotate": false,
    },
    "nice": true,
  },
  "xField": "sales",
  "yAxis": Object {
    "label": Object {
      "autoHide": true,
      "autoRotate": false,
    },
    "nice": true,
  },
  "yField": "type",
}
`;
